backend/dao/winesDAO.js

```diff
@@ -1,81 +1,56 @@
-function priceSelectors(filters) {
-  if (filters.price_eur) {
-    if (typeof filters.price_eur === "string") {
-      if (filters.price_eur == "low") {
-        query.price_eur = { $gte: 0, $lt: 20.0 };
-      } else if (filters.price_eur == "med") {
-        query.price_eur = { $gte: 20.0, $lt: 50.0 };
-      } else if (filters.price_eur == "high") {
-        query.price_eur = { $gte: 50, $lt: 75 };
-      } else if (filters.price_eur == "exp") {
-        query.price_eur = { $gte: 75 };
-      }
+function priceSelectors(priceFilter) {
+  let newPriceFilter = [];
+  if (priceFilter) {
+    if (typeof priceFilter === "string") {
+      newPriceFilter = [priceFilter];
     } else {
-      for (let i = 0; i < filters.price_eur.length; i++) {
-        if (filters.price_eur[i] == "low") {
-          singleFilter = { $gte: 0, $lt: 20.0 };
-        } else if (filters.price_eur[i] == "med") {
-          singleFilter = { $gte: 20.0, $lt: 50.0 };
-        } else if (filters.price_eur[i] == "high") {
-          singleFilter = { $gte: 50, $lt: 75 };
-        } else if (filters.price_eur[i] == "exp") {
-          singleFilter = { $gte: 75 };
-        }
-        filterObject = { price_eur: singleFilter };
-        //make it a list to prepare for query structure Mongodb accepts
-        filter_.push(filterObject);
-        console.log("filterObject: ", filterObject);
-        console.log("filter list: ", filter_);
+      newPriceFilter = [...priceFilter];
+    }
+    const priceQuery = [];
+    for (const price of newPriceFilter) {
+      if (price == "low") {
+        priceQuery.push({ price_eur: { $gte: 0, $lt: 20.0 } });
+      } else if (price == "med") {
+        priceQuery.push({ price_eur: { $gte: 20.0, $lt: 50.0 } });
+      } else if (price == "high") {
+        priceQuery.push({ price_eur: { $gte: 50, $lt: 75 } });
+      } else if (price == "exp") {
+        priceQuery.push({ price_eur: { $gte: 75 } });
       }
-      console.log("Before query assignment: ", query);
-      //query = { $or: filter_ };
-      //query.$and = [{ $or: filter_ }];
-
-      query1 = { $or: filter_ };
-      price_type.push(query1);
-      query.$and = price_type;
     }
-  }
 
-  filter_ = [];
-  singleFilter = {};
-  return query;
+    return priceQuery;
+  }
 }
-function foodSelectors(filters) {
-  if ("food_names" in filters) {
-    query.food_names = { $eq: filters["food_names"] };
+function foodSelectors(foodFilters) {
+  let foodNames;
 
-    return query;
-  }
+  foodNames = { $eq: foodFilters };
+
+  return foodNames;
 }
-function veganSelectors(filters) {
-  if (filters.vegan == "true") {
-    query.vegan = { $eq: true };
-  } else if (filters.vegan == "false") {
-    query.vegan = { $eq: false };
+function veganSelectors(veganFilters) {
+  let choice;
+  if (veganFilters == "true") {
+    choice = { $eq: true };
+  } else if (veganFilters == "false") {
+    choice = { $eq: false };
   } else {
     console.log("Invalid query");

```

Comment:
This should be an error.
Don't need to change now, but after the first iteration of the MVP, we should start considering the error messages returned to the client.